File: src/프로그래머스_Level1/숫자 문자열과 영단어.js
function solution(s) {
    return parseInt(
        s
            .replace(/zero/g, 0)
            .replace(/one/g, 1)
            .replace(/two/g, 2)
            .replace(/three/g, 3)
            .replace(/four/g, 4)
            .replace(/five/g, 5)
            .replace(/six/g, 6)
            .replace(/seven/g, 7)
            .replace(/eight/g, 8)
            .replace(/nine/g, 9),
    );
}

console.log(solution("one4seveneight"));
console.log(solution("23four5six7"));
console.log(solution("2three45sixseven"));
console.log(solution("123"));

File: src/프로그래머스_Level1/숫자 문자열과 영단어.py
def solution(s):
    return int(
        s.replace("zero", "0")
        .replace("one", "1")
        .replace("two", "2")
        .replace("three", "3")
        .replace("four", "4")
        .replace("five", "5")
        .replace("six", "6")
        .replace("seven", "7")
        .replace("eight", "8")
        .replace("nine", "9")
    )


print(solution("one4seveneight"))
print(solution("23four5six7"))
print(solution("2three45sixseven"))
print(solution("123"))
